src/sqlfluff/core/templaters/base.py

```diff
@@ -0,0 +1,359 @@
+"""Defines the templaters."""
+
+import logging
+from typing import Dict, Iterator, List, Tuple, Optional, NamedTuple
+
+_templater_lookup: Dict[str, "RawTemplater"] = {}
+
+# Instantiate the templater logger
+templater_logger = logging.getLogger("sqlfluff.templater")
+
+
+def templater_selector(s=None, **kwargs):
+    """Instantitate a new templater by name."""
+    s = s or "jinja"  # default to jinja
+    try:
+        cls = _templater_lookup[s]
+        # Instantiate here, optionally with kwargs
+        return cls(**kwargs)
+    except KeyError:
+        raise ValueError(
+            "Requested templater {0!r} which is not currently available. Try one of {1}".format(
+                s, ", ".join(_templater_lookup.keys())
+            )
+        )
+
+
+def register_templater(cls):

```

Comment:
Is this more related to making templaters "plugins" rather than "source mapping"?

Reply:
This is old code, that is just being moved around in this PR. I think this may get refactored as part of the plugin work, but that's a job for another day.